utils/crossref/parseDeposit.js

```diff
@@ -0,0 +1,79 @@
+const doiDataPaths = [
+	['book', 'content_item'],
+	['conference', 'conference_paper'],
+	['journal', 'journal_article'],
+	['posted_content'],
+];
+
+export const getDoiData = ({ doi_batch: { body } }) => {
+	let i = 0;
+
+	while (i < doiDataPaths.length) {
+		const path = doiDataPaths[i];
+		let visiting = body;
+		let j = 0;
+
+		while (j < path.length) {
+			const key = path[j];
+			visiting = visiting[key];
+
+			if (!visiting) {
+				break;
+			}
+
+			if (j === path.length - 1) {
+				return visiting.doi_data;
+			}
+
+			j++;
+		}
+
+		i++;
+	}
+
+	return null;
+};
+
+export const setDepositRecordContentVersion = (depositRecord, contentVersion) => {
```

Comment:
Okay, I think I understand your earlier comment in the reducer a little better now. It is odd that this function mutates its argument (right?) but then also returns it — it makes me think this should have a return type of `void` to make that explicit to the caller. But that makes it harder to use in the reducer...so, I don't have a concrete idea of what the best thing to do here is.

Reply:
I'll make it a void function that way it's clearer! Good idea.